vllm/utils.py

```diff
@@ -833,10 +854,40 @@ def make_tensor_with_pad(
 
     tensor = torch.from_numpy(padded_x).to(device)
     if pin_memory:
-        tensor = tensor.pin_memory()
+        if not current_platform.is_hpu():
+            tensor = tensor.pin_memory()
+        else:
+            tensor = tensor.pin_memory("hpu")
```

Comment:
can be removed, as it wont be called now

Reply:
Hi Michal, this method make_tensor_with_pad is still called from different places. It is replaced by make_tensor_with_pad_align in the repetition penaly. So I think we still need the check here.